src/MPQEditor/MPQCircleSelector.ts

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+
+import {
+  CircleGraphCtrl,
+  CircleGraphEvent,
+  MessageDefs,
+} from "../CircleGraph/CircleGraphCtrl";
+
+export interface MPQSelectionEvent {
+  onSelection(names: string[], document: vscode.TextDocument): void;
+}
+
+export type MPQSelectionCmdOpenArgs = {
+  modelPath: string;
+  document: vscode.TextDocument;
+  names: any;
+  viewColumn: vscode.ViewColumn | undefined;
```

Comment:
You may already know - below is possible.

```suggestion
  viewColumn?: vscode.ViewColumn;
```

Reply:
Thank you.